cmd/experimental/kjobctl/apis/v1alpha1/application_profile_types.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VolumeBundleReference is the name of the VolumeBundle.
+//
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+type VolumeBundleReference string
+
+type ApplicationProfileMode string
+
+const (
+	InteractiveMode ApplicationProfileMode = "Interactive"
+	JobMode         ApplicationProfileMode = "Job"
+)
+
+// +kubebuilder:validation:Enum=cmd;parallelism;completions;request;localqueue
+type Flag string
+
+const (
+	CmdFlag         Flag = "cmd"
+	ParallelismFlag Flag = "parallelism"
+	CompletionsFlag Flag = "completions"
+	RequestFlag     Flag = "request"
+	LocalQueueFlag  Flag = "localqueue"
+)
+
+// TemplateReference is the name of the template.
+// Template type depends on ApplicationProfileMode:
+//   - on Interactive mode it must be v1/PodTemplate
+//   - on Job mode it must be kjobctl.x-k8s.io/v1alpha1/JobTemplate
+//
```

Comment:
Maybe move this at L64, it's closer to ```Name ApplicationProfileMode `json:"name"` ``` and can make better sense.

Reply:
Done